slasher/src/config.rs

```diff
@@ -26,12 +27,19 @@ pub struct Config {
     pub history_length: usize,
     /// Update frequency in seconds.
     pub update_period: u64,
+    /// Offset from the start of the slot to begin processing.
+    #[serde(skip, default = "default_slot_offset")]
+    pub slot_offset: f64,
```

Comment:
The `serde(skip)` is a little bit of a hack to avoid a breaking database schema change. For some reason past Michael thought it would be a great idea to store the whole `Config` in the DB rather than just the three essential fields (the chunk sizes and the history length).

In part II of the slasher optimisations I'll clean this up, as I need to do a breaking DB schema change for #2286.

Reply:
Ah, unfortunately this broke the CLI test I wrote. Rather than introducing more config structs that will just be deleted I've weakened the test for now, and it can be re-strengthened in part II.